src/trainers/trainer.cpp

```diff
@@ -238,17 +238,31 @@ void trainer::evaluate(observer_ptr<model> model, execution_mode mode, El::Int n
 // Checkpointing
 // =============================================
 
-bool trainer::save_to_checkpoint_shared(persist& p) {
-  auto save_checkpoint = [&p](observer_ptr<execution_context> ctx)
-    ->void { ctx->save_to_checkpoint_shared(p); };
+bool trainer::save_to_checkpoint_shared() {
+  auto save_checkpoint = [this](observer_ptr<execution_context> ctx)
+    ->void { ctx->save_to_checkpoint_shared(this->get_persist_obj()); };
```

Comment:
```suggestion
  auto save_checkpoint = [this](observer_ptr<execution_context> ctx) {
    ctx->save_to_checkpoint_shared(this->get_persist_obj());
  };
```
Since you're just defining a new function, it should format the same way as functions.